crates/sui-framework/sources/vec_map.move

```diff
@@ -68,6 +82,36 @@ module sui::vec_map {
         &entry.value
     }
 
+    /// Return true if there is no overlap between the key sets of `self` and `other`.
+    public fun is_disjoint<K: copy, V>(self: &VecMap<K, V>, other: &VecMap<K, V>): bool {
+        let contents = &other.contents;
+        let i = 0;
+        let length = vector::length(contents);
+        while (i < length) {
+            let elem = vector::borrow(contents, i);
+            if (contains(self, &elem.key)) {
+                return false
+            };
+            i = i + 1
+        };
+        true
+    }
+
+    /// Merge the entries of `self` and `other`. Aborts if `self` and `other` have overlapping keys.
+    /// Afterward, `other` will be empty and can safely be destroyed.
+    public fun disjoint_union<K: copy, V>(self: &mut VecMap<K, V>, other: &mut VecMap<K, V>) {
```

Comment:
It is interesting to me that this API is the more natural choice in Move. Lots of fun learnings for an eventual `map` module/type

Reply:
Yeah... `other` being a reference instead of owned was interesting indeed. The owned API makes more sense, but I actually needed this one in my refactoring of `ValidatorSet`. We should probably have both (and similar for `vector::append`, which requires ownership for its equivalent of `other`).